src/prommis/examples/cmi_process_flowsheet/tests/test_cmi_process_flowsheet.py

```diff
@@ -0,0 +1,815 @@
+
+
+@pytest.mark.unit
+def test_build_flowsheet(system_frame):
+    m = system_frame
+    assert isinstance(m.fs, FlowsheetBlock)
+
+    # Check property packages
+    assert isinstance(m.fs.thermo_params, GenericParameterBlock)
+    assert isinstance(m.fs.dissolution_reaction_params, GenericReactionParameterBlock)
+    assert isinstance(m.fs.adjustment_reaction_params, GenericReactionParameterBlock)
+    assert isinstance(m.fs.precipitation_reaction_params, GenericReactionParameterBlock)
+
+    assert isinstance(m.fs.prop_gas, GenericParameterBlock)
+    assert isinstance(m.fs.prop_solid, PrecipitateParameters)
+    assert isinstance(m.fs.prop_liquid, AqueousParameter)
+
+    # Feed Stream (entering stage 1)
+    assert isinstance(m.fs.FEED, Feed)
+
+    # Copper(II) Nitrate Dissolution Stage (stage 1)
+    assert isinstance(m.fs.Dissolution, StoichiometricReactor)
+
+    # S/L Filter following dissolution (following stage 1)
+    assert isinstance(m.fs.S101, Separator)
+
+    # pH Adjustment Feed (entering stage 2)
+    assert isinstance(m.fs.AdjFeed, Feed)
+
+    # pH Adjustment Mixer (stage 2)
+    assert isinstance(m.fs.AdjMixer, Mixer)
+
+    # pH Adjustment Stage Reactor (stage 2)
+    assert isinstance(m.fs.Adjustment, StoichiometricReactor)
+
+    # Precipitation Reactant Feed Stream (stage 3)
+    assert isinstance(m.fs.PrecipFeed, Feed)
+
+    # Precipitation Mixer (stage 3)
+    assert isinstance(m.fs.PrecipMixer, Mixer)
+
+    # Precipitation Reactor (stage 3)
+    assert isinstance(m.fs.Precipitation, StoichiometricReactor)
+
+    # S/L Filter (following stage 3)
+    assert isinstance(m.fs.S102, Separator)
+
+    # Calcination Unit
+    assert isinstance(m.fs.Calcination, REEOxalateRoaster)
+
+    # Flowsheet connections
+    assert isinstance(m.fs.FEED_Diss, Arc)
+    assert isinstance(m.fs.Diss_S101, Arc)
+    assert isinstance(m.fs.S101_AdjMixer, Arc)
+    assert isinstance(m.fs.AdjFeed_AdjMixer, Arc)
+    assert isinstance(m.fs.AdjMixer_Adjustment, Arc)
+    assert isinstance(m.fs.Adjustment_PrecipMixer, Arc)
+    assert isinstance(m.fs.PrecipFeed_PrecipMixer, Arc)
+    assert isinstance(m.fs.PrecipMixer_Precipitation, Arc)
+    assert isinstance(m.fs.Precipitation_S102, Arc)
+
+
+@pytest.mark.component
+@pytest.mark.solver
+def test_degrees_of_freedom(system_frame):
+    m = system_frame
+    assert degrees_of_freedom(m) == 0
+
+
+@pytest.mark.component
+@pytest.mark.solver
+def test_solve(system_frame):
+    m = system_frame
+
+    results = solve_system(m)
+
+    assert_optimal_termination(results)
+
+
+@pytest.mark.component
+@pytest.mark.solver
+def test_numerical_issues(system_frame):
+    m = system_frame
+
+    dt = DiagnosticsToolbox(m)
+    dt.assert_no_numerical_warnings()
+
+
+@pytest.mark.component
+@pytest.mark.solver
+def test_solution(system_frame):
+    m = system_frame
+
+    #  FEED (entering Stage 1)
+    assert m.fs.FEED.outlet.flow_mol_phase_comp[0, "Liq", "H2O"].value == pytest.approx(
```

Comment:
Please check that these all check `pytest.approx([value], rel=1e-4`, if not please specify that explicitly. If the value is zero then you'll have to use `abs-1e-4`.